lib/download-file.js

```diff
@@ -39,7 +40,12 @@ function downloadFileAsync(file, options = {}) {
       new Promise(x => fileClose = x),
       new Promise(x => responseEnd = x),
     ];
-    const request = req.get(file, (response) => {
+
+    let reqOptions = { agent: proxy.proxyEnabled() ? proxy.proxyAgent : req.globalAgent };
```

Comment:
let's move this logic into a class `HttpsAgent` and have it calculate the Proxy or fallback to the default agent